src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -5,10 +5,45 @@
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MINIMAL_VALID_AGE = 18;
+    private static final int MINIMAL_VALID_PASSWORD_LENGTH = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        checkUser(user);
+        checkLogin(user.getLogin());
+        checkAge(user);
+        checkPassword(user.getPassword());
+        return storageDao.add(user);
+    }
+
+    private void checkLogin(String login) {
+        if (storageDao.get(login) != null) {
+            throw new ValidationException("Invalid data. User with this login already exists");
+        } else if (login == null || login.length() == 0) {
```

Comment:
```suggestion
        }
        if (login == null || login.isEmpty()) {
```
this check should be the first one